### PR TITLE
Update 0.9, multifaction

### PR DESCRIPTION
**Pull request by Zetrith**
Includes implementation of multifaction beta. The commit message has more details

Some small things are still missing, I'll add them soon:
- Multifaction hosting option is unused
- You can't choose a new faction's ideology

### CODE REVIEW

**Review by notfood — COMMENTED** [review]
I've gone through the changes, a lot of the Faction part is flying over my head. I'll need to experiment with it to see how it impacts mod compatibility.

File: Source/Client/Factions/MultifactionPatches.cs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+[HarmonyPatch(typeof(MapParent), nameof(MapParent.CheckRemoveMapNow))]
+static class CheckRemoveMapNowPatch
+{
+    static bool Prefix(MapParent __instance)
+    {
+        return __instance.Faction is not { IsPlayer: true };
+    }
+}
+
+// todo this is temporary
```
[the remark this answers is not in the document]
Comment (Zetrith) [reply Source/Client/Factions/MultifactionPatches.cs]: Removed

File: Source/Client/EarlyInit.cs [hunk 2]
```diff
@@ -66,18 +66,18 @@ internal static void InitSync()
...
-            new HarmonyMethod(typeof(ThingCategoryDef_ThisAndChildCategoryDefsPatch), "Prefix"),
-            new HarmonyMethod(typeof(ThingCategoryDef_ThisAndChildCategoryDefsPatch), "Postfix")
-        );
+        // harmony.PatchMeasure(
+        //     AccessTools.Method(typeof(ThingCategoryDef), "get_ThisAndChildCategoryDefs"),
+        //     new HarmonyMethod(typeof(ThingCategoryDef_ThisAndChildCategoryDefsPatch), "Prefix"),
+        //     new HarmonyMethod(typeof(ThingCategoryDef_ThisAndChildCategoryDefsPatch), "Postfix")
+        // );
 
         if (MpVersion.IsDebug)
```
[the remark this answers is not in the document]
Comment (Zetrith) [reply Source/Client/EarlyInit.cs:68]: This is called from threaded code now which complicates the caching optimization and makes it obsolete anyway probably. Removed

File: Source/Client/Debug/DebugPatches.cs [hunk 3]
```diff
@@ -177,8 +177,8 @@ static void Prefix(ref string text)
         {
```
[the remark this answers is not in the document]
Comment (Zetrith) [reply Source/Client/Debug/DebugPatches.cs:177]: Left it in by mistake

**Comment by SokyranTheDragon** [comment]
I've looked through most the changes. There were many additions/changes that were rather small and simple to understand, or removal of stuff I assumed are no longer needed - I mostly skimmed through those. As for the bigger changes, I'll have to look deeper into them.

**Pull request merged**